mautic/mautic
# DPMMA-2831 Unsubscribe from all in preferences center

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | https://github.com/mautic/user-documentation/pull/332
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
When the preference center is activated, there is no option for users to unsubscribe from all marketing messages. This PR allows users to set the "Do Not Contact" flag by clicking a link in the subscription center.
![image](https://github.com/user-attachments/assets/51453181-ca1d-4cad-a7ad-d180541a8ccf)

Additionally, a new token is available to use in emails: `{dnc_url}`, so you can create an email footer like this:
![image](https://github.com/user-attachments/assets/48520b4c-928b-483c-b7a6-e17563859574)






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Activate Preferences center Mautic settings
3. Send email to user with `unsubscribe_url` and `dnc_url`:
 ```
 Manage your <a href="{unsubscribe_url}">email preferences</a> or <a href="{dnc_url}">unsubscribe from all marketing messages</a>⁠⁠⁠⁠⁠⁠⁠.
 ```
4. Open the email and click the email preferences link.
5. Click the "Unsubscribe from all marketing messages" link in the preferences center.
6. Check if the contact is unsubscribed.
7. Click the resubscribe link.
8. Check if the contact is resubscribed.
9. Click the direct unsubscribe link in the email.
10. Check if the contact is unsubscribed.
